### PR TITLE
Fix JIRA update issue if issue already linked

### LINKED ISSUE #2438: Issues that have a JIRA link update, even if "push to jira" not ticked
**Bug description**
When updating the status of a finding that already has a JIRA link to it, it will go update the JIRA ticket status, even though the "push to JIRA" checkbox is not checked, and even though the product is not configured to "push all to JIRA".

FYI @Apipia  

**Steps to reproduce**

**Expected behavior**
I have control over what gets pushed to JIRA and don't want surprises.

**Deployment method** *(select with an `X`)*
- [ ] Kubernetes
- [x] Docker
- [ ] setup.bash / legacy-setup.bash

**Environment information**
`5bca73ba: Merge remote-tracking branch 'refs/remotes/origin/dev' into dev [2020-05-27 07:48:25 +0000]`

### PR DESCRIPTION
**Pull request by madchap**
Fix #2438 and #2440

### CODE REVIEW

File: dojo/finding/views.py [hunk 1]
```diff
@@ -665,11 +666,13 @@ def edit_finding(request, fid):
...
+            if push_all_issues_enabled:
                 push_to_jira = True
-            elif 'jiraform-push_to_jira' in request.POST:
-                jform = JIRAFindingForm(request.POST, prefix='jiraform', enabled=enabled)
+            # elif 'jiraform-push_to_jira' in request.POST:
+            elif request.POST['jiraform-push_to_jira']:
+                jform = JIRAFindingForm(request.POST['jiraform-push_to_jira'], prefix='jiraform', enabled=True)
                 if jform.is_valid():
+                    # push_to_jira = True  ## easy way out but why?
                     push_to_jira = jform.cleaned_data.get('push_to_jira')
```
Comment (madchap) [inline dojo/finding/views.py:676]: Any idea why this always returns `False` even if the checkbox is ticked?

Comment (valentijnscholten) [reply dojo/finding/views.py:676]: Would you need to replace `request.POST['jiraform-push_to_jira']` by `request.POST`?

Comment (valentijnscholten) [reply dojo/finding/views.py:676]: oh I see on slack you already solved it?

Comment (madchap) [reply dojo/finding/views.py:676]: No. Either one, result seems to be the same.

**Comment by madchap** [comment]
My manual tests confirm my "basic" issue is fixed with this, even though I have to resort to a band-aid on `jform.cleaned_data.get('push_to_jira')`, which always returns `False` for some reasons.

For the other issues mentioned by @dayal18 at https://github.com/DefectDojo/django-DefectDojo/issues/2438, I suggest we work on another PR and keep things moving with small PRs.

**Comment by madchap** [comment]
Rebased `--onto` release/1.6.1.

**Comment by dayal18** [comment]
I tested these changes in my local.
And they are working just fine. :) If this helps. :)

**Pull request merged**